libbeat/cmd/instance/beat_test.go

```diff
@@ -269,3 +271,79 @@ func (r *outputReloaderMock) Reload(
 	r.cfg = cfg
 	return nil
 }
+
+func TestPromoteOutputQueueSettings(t *testing.T) {
+	tests := map[string]struct {
+		input                 []byte
+		memEvents             int
+		expectValidationError bool
+	}{
+		"blank": {input: []byte(""),
+			memEvents: 4096},
+		"defaults": {input: []byte(`
+name: mockbeat
+output:
+  elasticsearch:
+    hosts:
+      - "localhost:9200"
+`),
+			memEvents: 4096},
+		"topLevelQueue": {input: []byte(`
+name: mockbeat
+queue:
+  mem:
+    events: 8096
+output:
+  elasticsearch:
+    hosts:
+      - "localhost:9200"
+`),
+			memEvents: 8096},
+		"outputLevelQueue": {input: []byte(`
+name: mockbeat
+output:
+  elasticsearch:
+    hosts:
+      - "localhost:9200"
+    queue:
+      mem:
+        events: 8096
```

Comment:
I assume the disk queue can also be enabled this way? Do we want to enable configuring the disk queue for agent? Or should we have an explicit check that if a Beat is managed by agent the disk queue is disabled.

I lean towards the latter because we haven't planned any testing effort around the disk queue in agent yet, and the goal of this implementation was mostly to allow performance tuning of the memory queue settings in the field.

I could be convinced otherwise.

Reply:
> I assume the disk queue can also be enabled this way?

Yep

> Do we want to enable configuring the disk queue for agent? Or should we have an explicit check that if a Beat is managed by agent the disk queue is disabled.
> 
> I lean towards the latter because we haven't planned any testing effort around the disk queue in agent yet, and the goal of this implementation was mostly to allow performance tuning of the memory queue settings in the field.
> 
> I could be convinced otherwise.

I'll code up a check and we can evaluate.  My personal preference is not for Beats to have different features when run under agent.  It makes it harder to debug if the the same config does different things when run under agent or not.  But given that users can input any yaml, this might be acceptable.

Reply:
take a look at `checkAgentDiskQueue` in beat.go

Reply:
we could enable disk queue as a tech preview or beta and perform the testing at a later sprint. for the most part the functionality has been in beats for a while.

Reply:
Actually I've changed my mind on this, the disk queue needs changes in Elastic Agent to work properly so we shouldn't allow using it.

Agent needs to create a unique disk queue path per running Beat, or all of them will try to share the same queue which would lead to data loss or duplication. We also need to put the disk queue outside of the agent data directory by default so it doesn't have to be copied between upgrades. This isn't as simple as allowing us to turn it on in the configuration.

I will create a follow up issue specifically for this allowing this. For now let's forbid using the disk queue under agent to avoid allowing configurations that don't work properly.

Reply:
See https://github.com/elastic/elastic-agent/issues/3490 for the complications involved in enabling the disk queue for the Elastic Agent.

Reply:
thanks for looking into this Craig.